core/modules/tiddler.js

```diff
@@ -12,6 +12,39 @@ Extension methods for the $tw.Tiddler object (constructor and methods required a
 /*global $tw: false */
 "use strict";
 
+/*
+returns names of tiddler fields
+
+default: the current tiddler's custom fields, or via...
+  => listFields("$custom")
+all fields:
+  => listFields("$all")	
```

Comment:
> It's a bit inconsistent that $text and $standard return their field names regardless of whether they are present on this particular tiddler, versus the other options that only return fields that are defined on the tiddler.

I agree, I modified the code so only existing core fields are returned. Would be nice to have a proper interface though to read the core fields from, "text" or "all standard"... rather than have them declared as variables in this function.

> tiddler.getFieldNames() that returns a classified subset of the fields on the specified tiddler
> wiki.getFieldNames() that returns the full set of field names in each category

@Jermolene: I agree.

1. implemented both
2. used in the code
3. updated the demo
4. pushed in a commit